jobapi/middleware_test.go

```diff
@@ -0,0 +1,100 @@
+package jobapi_test
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"net/http/httptest"
+	"testing"
+
+	"github.com/buildkite/agent/v3/jobapi"
+	"github.com/google/go-cmp/cmp"
+)
+
+func testHandler(w http.ResponseWriter, r *http.Request) {
+	w.WriteHeader(http.StatusOK)
+	w.Write([]byte("{}"))
+}
+
+func TestAuthMdlw(t *testing.T) {
+	t.Parallel()
+
+	token := "llamas"
+	cases := []struct {
+		title    string
+		auth     string
+		wantCode int
+		wantBody map[string]string
+	}{
+		{
+			title:    "valid token",
+			auth:     fmt.Sprintf("Bearer %s", token),
+			wantCode: http.StatusOK,
+			wantBody: map[string]string{},
+		},
+		{
+			title:    "invalid token",
+			auth:     "Bearer alpacas",
+			wantCode: http.StatusUnauthorized,
+			wantBody: map[string]string{"error": "invalid authorization token"},
+		},
+		{
+			title:    "non-bearer auth",
+			auth:     fmt.Sprintf("Basic %s", token),
+			wantCode: http.StatusUnauthorized,
+			wantBody: map[string]string{"error": "invalid authorization header: type must be Bearer"},
+		},
+		{
+			title:    "no auth",
+			auth:     "",
+			wantCode: http.StatusUnauthorized,
+			wantBody: map[string]string{"error": "authorization header is required"},
+		},
+	}
+
+	for _, c := range cases {
+		c := c
+		t.Run(c.title, func(t *testing.T) {
+			t.Parallel()
```

Comment:
do the child test cases need to be `t.Parallel`ed if the [parent test is already parallel?](https://github.com/buildkite/agent/pull/1943/files#diff-6d895c7a60f16288c48e247e795f18cb4634b9b1def94d2bec65fc8c67885e1bR20)

Reply:
If you want them to be parallel with each other, then I believe so: https://pkg.go.dev/testing#T.Run